### PR TITLE
fix(gatsby): fix --https option for develop

### LINKED ISSUE #36241: --https doesn't work since Gatsby 4.19.0
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

The [Local HTTPS Option](https://www.gatsbyjs.com/docs/local-https/) (`--https`) doesn't work anymore in Gatsby Version 4.19.0, 4.19.1 and 4.19.2.
The terminal still shows:

> You can now view my-gatsby-site in the browser.
>   https://localhost:8000/

But in the browser, only `http://localhost:8000/` works.

This is reproducable in gatsby-starter bundle, by simply appending `--https --key-file ....key --cert-file ....cert` to the start-script.

### Reproduction Link

https://github.com/jens-duttke/gatsby-https-fault

### Steps to Reproduce

1. git clone https://github.com/jens-duttke/gatsby-https-fault.git
2. cd gatsby-https-fault
3. npm i
4. npm start
5. Go to https://localhost:8000/ as shown in the terminal

### Expected Result

The page will be shown

### Actual Result

ERR_SSL_PROTOCOL_ERROR

### Environment

```shell
Windows 10 Pro, Gatsby 4.19.x
```

### Config Flags

None

### PR DESCRIPTION
**Pull request by pieh**
## Description

https://github.com/gatsbyjs/gatsby/pull/35675 broke https option, this change restores it

## Related Issues

Fixes #36241

### CODE REVIEW

File: packages/gatsby/src/commands/types.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ import { Store, AnyAction } from "redux"
 import { IGatsbyState } from "../redux/types"
 
 export interface ICert {
-  key: Buffer
-  cert: Buffer
+  key: string
+  cert: string
```
Comment (pieh) [inline packages/gatsby/src/commands/types.ts:7]: we are passing ICert Object to child process, so this has to be serializable. Buffer can be serialized but but this was fastest way to fix as Strings are also supported.

Without this I was getting:
```
The "options.cert" property must be of type string or an instance of Buffer, TypedArray, or DataView. Received an instance of Object
```

**Review by LekoArts — APPROVED** [review]

**Comment by badaczewski** [comment]
Any idea when this will get a release? This is broken in 4.19.0, 4.19.1 and 4.19.2.

**Comment by jjang16** [comment]
This issue is pretty urgent. Build is broken from 4.18 and below, and Dev is broken 4.19

**Comment by christopherrobinson** [comment]
Can we get a release with this fix in? Currently hacking this in with with [patch-package](https://www.npmjs.com/package/patch-package)

**Comment by LekoArts** [comment]
A minor release goes out every two weeks, so the next one is tomorrow.

**Comment by pieh** [comment]
This was released in `gatsby@next`, so you can use that temporarily (it will be released in `gatsby@latest` stable tomorrow)

**Pull request merged**